packages/visx-brush/src/BaseBrush.tsx

```diff
@@ -33,11 +34,17 @@ export type BaseBrushProps = {
   clickSensitivity: number;
   disableDraggingSelection: boolean;
   resetOnEnd?: boolean;
+  isUseWindowMoveEvents?: boolean;
 };
 
 export type BaseBrushState = BrushShape & {
   activeHandle: ResizeTriggerAreas | null;
   isBrushing: boolean;
+  isDragInProgress: boolean;
```

Comment:
would it be possible to use the existing `isBrushing` flag instead of introducing `isDragInProgress` and `isMovingBrushSelection`? it's not clear readability-wise why we have 3 (will keep reading but that's my initial impression)

Reply:
Next week I'll try solution with one enum variable

Reply:
Now it should be much easier to understand. I replaced all new boolean variables with bushing type which help to understand what kind of action in progress. It helps me in moveMove event listener to understand how the move should be handled (for example it slightly different for moving brushing area and selecting a new one).